source/standard_scores.Rmd

```diff
@@ -0,0 +1,1515 @@
+pct_diff = (demo - repub) / repub * 100
+```
+
+In fact, if we look at all 441 congressional districts in Kaptur's table, we
+find a large difference in the average median household income for Democrat and
+Republican districts; the Democrat districts are, on average, about
+`r round(py$pct_diff)`% richer (@tbl-party_means).
+
+```{r echo=FALSE, eval=TRUE}
+means <- py$means
+means$Median_Income <- formattable::currency(means$Median_Income, digits = 0)
+names(means) <- "Mean of median household income"
+ketable(means,
+        caption = "Means for median household income by party {#tbl-party_means}")
+```
+
+Next we are going to tip our hand, and show how we got these data.  In previous
+chapters, we had {{< var cell >}}s like this in which we enter the values we
+will analyze.  These example values are from @sec-public-liquor:
+
+```{python}
+# Liquor prices for US states with private market.
+priv = np.array([
+    4.82, 5.29, 4.89, 4.95, 4.55, 4.90, 5.25, 5.30, 4.29, 4.85, 4.54, 4.75,
+    4.85, 4.85, 4.50, 4.75, 4.79, 4.85, 4.79, 4.95, 4.95, 4.75, 5.20, 5.10,
+    4.80, 4.29])
+```
+
+```{r}
+# Liquor prices for US states with private market.
+priv <- c(4.82, 5.29, 4.89, 4.95, 4.55, 4.90, 5.25, 5.30, 4.29, 4.85, 4.54,
+          4.75, 4.85, 4.85, 4.50, 4.75, 4.79, 4.85, 4.79, 4.95, 4.95, 4.75,
+          5.20, 5.10, 4.80, 4.29)
+```
+
+Now we have 441 values to enter, and it is time to introduce {{< var lang >}}s
+standard tools for loading data.
+
+### Comma-separated-values (CSV) format {#sec-csv-format}
+
+The data we will load is in a file on disk called `data/congress_2023.csv`.
+These are data from Kaptur's table in a comma-separated-values (CSV) format
+file. We refer to this file with its *filename*, containing the directory
+(`data/`) followed by the name of the file (`congress_2023.csv`), giving
+a filename of `data/congress_2023.csv`.
+
+The *CSV format* is a very simple text format for storing table data.  Usually,
+the first line of the CSV file contains the column names of the table, and the
+rest of the lines contain the row values.  As the name suggests, commas (`,`)
+separate the column names in the first line, and the row values in the
+following lines. If you opened the `data/congress_2023.csv` file in some editor,
+such as Notepad on Windows or TextEdit on Mac, you would find that the first few
+lines looked like this:
+
+```{python echo=FALSE, eval=TRUE, results="asis"}
+from pathlib import Path
+backticks = '```'
+lines = '\n'.join(
+    Path('data/congress_2023.csv').read_text().splitlines()[:6])
+# Reticulate seems to suffer with the literal backticks here.
+print(f"""{backticks}
+{lines}
+{backticks}
+""")
+```
+
+::: python
+
+### Introducing the Pandas library {#sec-pandas-intro}
+
+Here we start using the Pandas library to load table data into Python.
+
+Thus far we have used the Numpy library to work with data in arrays.  Pandas is
+As always with Python, when we want to use a library, we have to `import` it
```

Comment:
As always with Python, when we want to use a library like Pandas, we have to `import` it.